Support for bridge tcp user timeout option

abiliojr opened the pull request:
Please find the description of this change in #2230

- [x] Have you signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php), using the same email address as you used in your commits?
- [x] Do each of your commits have a "Signed-off-by" line, with the correct email address? Use "git commit -s" to generate this line for you.
- [ ] If you are contributing a new feature, is your work based off the develop branch?
- [ ] If you are contributing a bugfix, is your work based off the fixes branch?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you successfully run `make test` with your changes locally?

ralight commented:
This looks fine to me in principle, but I think it needs a bit more work.

* You disallow values == 0, but then seem to apply a value of zero if the option isn't set.
* I'd like to see more documentation around what this means, even if it is pointing to somewhere to read something else. Ideally some hints about what might be appropriate would be good.
* If this option isn't used, what is the OS default value? Without knowing that, or being able to find it out I don't know whether I need to reduce the value, or what it is reasonable to reduce it to.

I think the idea is good, it's just a bit opaque at the moment and so "nobody" would use it.

abiliojr commented:
>     * You disallow values == 0, but then seem to apply a value of zero if the option isn't set.
Changed to allow values of 0 (which translates to user default) or don't set the socket option by default.

>     * I'd like to see more documentation around what this means, even if it is pointing to somewhere to read something else. Ideally some hints about what might be appropriate would be good.
I added some extra information in the documentation/config example. Please let me know if it's enough.

>     * If this option isn't used, what is the OS default value? Without knowing that, or being able to find it out I don't know whether I need to reduce the value, or what it is reasonable to reduce it to.
Added a reference to the linux man page that specifies up to 20 minutes by default. This is the consequence of the default value of _tcp_retries2_ (15). Mentioned this relation in the mosquitto documentation.

I also added a [link to a page](https://blog.cloudflare.com/when-tcp-sockets-refuse-to-die/) explaining how this setting works, how to use it, and it's caveats.

@ralight , please let me know if these changes are enough
